src/geojson.rs

```diff
@@ -0,0 +1,189 @@
+use std::{
+    collections::{BTreeMap, HashMap, HashSet},
+    io::Write,
+    io::{self, BufWriter},
+};
+
+use anyhow::{bail, Context, Result};
+use geojson::{self, GeoJson, Geometry, JsonObject, JsonValue};
+use osmpbfreader::{OsmId, OsmObj, Ref, Way};
+
+use crate::{filter, util::FloatTuple};
+
+pub fn write(objs: &BTreeMap<OsmId, OsmObj>, out: impl io::Write) -> Result<()> {
+    // Use a buffered writer to amortize flushes.
+    let mut buffer = BufWriter::new(out);
+
+    for relation in objs.values().filter(|obj| filter::by_target(obj)) {
+        if let Some(feature) = to_feature(relation, objs) {
+            let serialized = feature.to_string();
+            writeln!(buffer, "{serialized}")?;
+        }
+    }
+
+    Ok(())
+}
+
+fn to_feature(obj: &OsmObj, all_objs: &BTreeMap<OsmId, OsmObj>) -> Option<geojson::GeoJson> {
+    let tags = obj.tags();
+    let name = tags.get("name")?;
+    let admin_level = tags.get("admin_level")?;
+    let ars = tags.get("de:regionalschluessel")?;
+
+    let mut properties = JsonObject::new();
+    properties.insert(
+        String::from("adminLevel"),
+        JsonValue::from(admin_level.parse::<u8>().ok()?),
+    );
+    properties.insert(String::from("ars"), JsonValue::from(ars.as_str()));
+    properties.insert(String::from("name"), JsonValue::from(name.as_str()));
+    properties.insert(
+        String::from("osm:id"),
+        JsonValue::from(obj.relation()?.id.0),
+    );
+    properties.insert(String::from("osm:type"), JsonValue::from("relation"));
```

Comment:
Mit `json!` könntest du dir eine Menge type conversions ersparen (und ohne `mut` arbeiten). Weil `json!` ein [`serde_json::Value`](https://docs.rs/serde_json/latest/serde_json/value/enum.Value.html) zurückgibt, du aber weisst, dass du eine `serde_json::Map` konstruierst und brauchts, kannst du dir mit pattern matching (hier via [let-else](https://doc.rust-lang.org/beta/rust-by-example/flow_control/let_else.html)) die richtige variant holen und ansonsten bailen,

```suggestion
    let serde_json::Value::Object(properties) = json!({
        "adminLevel":admin_level.parse::<u8>().ok()?,
        "ars": ars,
        "name": name,
        "osm:id": obj.relation()?.id.0,
        "osm:type": "relation",
    }) else {
        return None;
    };
```